lib/github.ml

```diff
@@ -36,6 +36,21 @@ module Parse = struct
     | _ when Bos_setup.String.is_prefix uri ~affix:"https://" ->
         user_from_regexp_opt uri "https://github\\.com/\\(.+\\)/.+\\(\\.git\\)?"
     | _ -> None
+
+  let path_from_regexp_opt uri regexp =
+    try
+      Some
+        ( "git@github.com:"
+        ^ Re.(Group.get (exec (Emacs.compile_pat regexp) uri) 1) )
+    with Not_found -> None
+
+  let ssh_uri_from_http uri =
+    match uri with
+    | _ when Bos_setup.String.is_prefix uri ~affix:"git@" ->
+        path_from_regexp_opt uri "git@github\\.com:\\(.+\\)"
```

Comment:
Is it really necessary here? we're basically reconstructing the same URI, can't we just return it as it is?

Reply:
This is required for something this PR adds:
It prints a warning when guessing the URI is needed but didn't work.